src/NuGet.Core/NuGet.Protocol.VisualStudio/ExtensibleSourceRepositoryProvider.cs

```diff
@@ -84,17 +87,19 @@ public Configuration.IPackageSourceProvider PackageSourceProvider
             get { return _packageSourceProvider; }
         }
 
-        private void Init()
+        private void ResetRepositories()
         {
-            _repositories.Clear();
+            var repositories = new List<SourceRepository>();
             foreach (var source in _packageSourceProvider.LoadPackageSources())
             {
                 if (source.IsEnabled)
                 {
                     var sourceRepo = new SourceRepository(source, _resourceProviders);
-                    _repositories.Add(sourceRepo);
+                    repositories.Add(sourceRepo);
                 }
             }
+
+            Interlocked.CompareExchange(ref _repositories, repositories, null);
```

Comment:
`LoadPackageSources` could raise PackageSourceChanged event which will again call into this Init api, so this is just to be more defensive. Although I never saw it raising that event through that API but then assigning _repositories at the end of Init call is also some kind of defensive approach.